frontend/src/pages/team/Devices/index.vue

```diff
@@ -95,36 +102,52 @@ export default {
     mixins: [permissionsMixin],
     data () {
         return {
-            loading: false,
+            loading: true,
             creatingDevice: false,
             deletingDevice: false,
-            devices: [],
-            checkInterval: null
+            devices: new Map(),
+            checkInterval: null,
+            nextCursor: null
         }
     },
     watch: {
         team: 'fetchData'
     },
     mounted () {
-        // Set loading flag to true for initial page load
-        this.loading = true
-        this.fetchData()
-        this.checkInterval = setInterval(() => {
-            // Do not set loading flag so the refresh happens in the background
-            this.fetchData()
-        }, 10000)
+        this.pollForData()
     },
     unmounted () {
         clearInterval(this.checkInterval)
     },
     methods: {
-        fetchData: async function (newVal) {
-            if (this.team.id) {
-                const data = await teamApi.getTeamDevices(this.team.id)
-                this.devices.length = 0
-                this.devices = data.devices
+        async pollForData () {
+            try {
+                if (this.team.id) {
+                    await this.fetchData(null, true) // to-do: For now, this only polls the first page...
```

Comment:
I want to call this out explicitly: As implemented now, polling is only updating the first page of results. Given server side updates is in the roadmap, it didn't seem worth entering this rabbit hole for what should be a bug-fix PR. 

I'll spin up an issue to either re-think polling on this page, or remove this logic once server side events is in place. For now, it should be considered a known limitation of the devices page.

Reply:
Why would we want updates from devices that are not on the rendered page?

While I understand we're accruing technical debt, it seems I'm missing something implicit in your comment why it's an issue?

Reply:
The devices table (and data-table) doesn't currently have pages or page numbers visually. The load more button simply loads the next page from the API and appends it to the bottom of the table. Perhaps the comment could be clearer.